### PR TITLE
Instead of using the (nil) err, use the non-nil newCtErr for ctErr.

### PR DESCRIPTION
**Pull request by ncabatoff**

### CODE REVIEW

**Comment by ncabatoff** [comment]
Looks like it would result in a nil pointer panic on line 608 when we call ctErr.Error(), if checkNeedsCG returned a non-nil error.

File: vault/request_handling.go [hunk 1]
```diff
@@ -565,7 +565,7 @@ func (c *Core) handleRequest(ctx context.Context, req *logical.Request) (retResp
 		newCtErr, cgResp, cgAuth, cgRetErr := checkNeedsCG(ctx, c, req, auth, ctErr, nonHMACReqDataKeys)
 		switch {
 		case newCtErr != nil:
-			ctErr = err
+			ctErr = newCtErr
```
Comment (briankassouf) [inline vault/request_handling.go]: I also wonder if on line 573 below we should be returning cgRetErr instead of retErr

**Comment by briankassouf** [comment]
This does seem like a bug, but we should really test this change out well

**Comment by briankassouf** [comment]
Whoops, didn't mean to close it

File: vault/request_handling.go (hunk 1 above)
Comment (jefferai) [reply vault/request_handling.go]: Dug into this more. I think my confusion is that this code was different before The Big One. This is indeed wrong, however, newCtErr will only ever be non-nil if there is an issue creating the control group token, and that will only run if ctErr is a single error and that one error is that it needs a control group created.

@briankassouf I wonder if we should actually simplify the checkNeedsCG function on the enterprise side quite a bit by moving up the multierror error count/type check. That will also make it more clear here what the error swapping is all about.

**Pull request merged**